### PR TITLE
Web analytics

### PR DESCRIPTION
**Pull request by jprochazk**
### What

Part of https://github.com/rerun-io/rerun/issues/2966

- Enable analytics on the web with the same events as native viewer
  - Events have a `wasm32-unknown-unknown` target, and have the url set to the string value of `window.location`
- Expose `set_email` global function when running on the web
  - Used to identify Rerun employees in the same way as `rerun analytics email <email@rerun.io>`
- Disable analytics if the web viewer is running in the context of a notebook
  - To detect it, we set the `notebook=1` query param on the iframe

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3166) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3166)
- [Docs preview](https://rerun.io/preview/f03ef1a8af8cbe2e37541f105a4b292da61ef448/docs) 
- [Examples preview](https://rerun.io/preview/f03ef1a8af8cbe2e37541f105a4b292da61ef448/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

**Comment by emilk** [comment]
> Expose `set_email` global function when running on the web

We should document how to run this, and who should run this. `rerun analytics email` is currently documented [here](https://www.notion.so/rerunio/Onboarding-8a9d2350d5464f598e1e5a01cc91eea6?pvs=4#9f8adef9b8c34f6d88dcc3a7361244c1), so I guess it makes sense to document `set_email` in the same place (no need to make it public, even though it is hardly a secret)

File: crates/re_viewer/src/web.rs [hunk 1]
```diff
@@ -143,6 +144,14 @@ fn create_app(cc: &eframe::CreationContext<'_>, url: Option<String>) -> crate::A
     app
 }
 
+#[cfg(feature = "analytics")]
+#[wasm_bindgen]
+pub fn set_email(email: String) {
```
Comment (emilk) [inline crates/re_viewer/src/web.rs:149]: This deserves a docstring

**Review by emilk — CHANGES_REQUESTED** [review]
Looks good! But we really need to limit what urls we send to analytics 😬

**Comment by jprochazk** [comment]
> rerun analytics email is currently documented [here](https://www.notion.so/rerunio/Onboarding-8a9d2350d5464f598e1e5a01cc91eea6?pvs=4#9f8adef9b8c34f6d88dcc3a7361244c1), so I guess it makes sense to document set_email in the same place

Yeah, I wasn't sure where exactly to put it. I updated the document.

File: crates/re_analytics/src/web/config.rs [hunk 2]
```diff
@@ -20,34 +22,77 @@ pub enum ConfigError {
...
 
-// NOTE: all the `rename` clauses are to avoid a potential catastrophe :)
 #[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
 pub struct Config {
-    #[serde(rename = "analytics_enabled")]
-    pub analytics_enabled: bool,
-
     // NOTE: not a UUID on purpose, it is sometimes useful to use handcrafted IDs.
-    #[serde(rename = "analytics_id")]
+    #[serde(rename = "i")]
```
[the remark this answers is not in the document]
Comment (jprochazk) [reply crates/re_analytics/src/web/config.rs:28]: These are completely arbitrary. I thought keeping the size down may matter here, but on secound thought it almost certainly doesn't. I'll rename them back.

File: scripts/ci/demo_assets/static/index.js [hunk 3]
```diff
@@ -56,12 +56,12 @@ function load_wasm() {
...
 
-  const status_element = document.getElementById('status');
+  const status_element = document.getElementById("status");
   function progress({ loaded, total_bytes }) {
     if (total_bytes != null) {
-      status_element.innerHTML = Math.round(Math.min(loaded / total_bytes * 100, 100)) + '%';
+      status_element.innerHTML = Math.round(Math.min((loaded / total_bytes) * 100, 100)) + "%";
     } else {
-      status_element.innerHTML = (loaded / (1024 * 1024)).toFixed(1) + 'MiB'
+      status_element.innerHTML = (loaded / (1024 * 1024)).toFixed(1) + "MiB";
```
[the remark this answers is not in the document]
Comment (jprochazk) [reply scripts/ci/demo_assets/static/index.js:64]: I have all `.js`/`.ts`/`.css`/`.html` files default to prettier in my global vscode config, we can add it: https://github.com/rerun-io/rerun/issues/3180

File: crates/re_viewer/src/viewer_analytics.rs [hunk 4]
```diff
@@ -23,11 +25,8 @@ pub struct ViewerAnalytics {
...
 impl ViewerAnalytics {
     #[allow(unused_mut, clippy::let_and_return)]
-    pub fn new(
-        disabled: bool,
-        #[cfg(target_arch = "wasm32")] analytics_url: Option<String>,
-    ) -> Self {
-        if disabled {
+    pub fn new(startup_options: &StartupOptions) -> Self {
+        if startup_options.is_in_notebook {
             return Self { analytics: None };
```
Comment (emilk) [inline crates/re_viewer/src/viewer_analytics.rs:30]: We should explain here why we return early in notebooks

File: crates/re_viewer/src/viewer_analytics.rs [hunk 5]
```diff
@@ -42,8 +41,10 @@ impl ViewerAnalytics {
         let mut analytics = Self { analytics };
 
         #[cfg(target_arch = "wasm32")]
-        if let Some(url) = analytics_url {
-            analytics.register("url", url);
+        if let Some(location) = startup_options.location.as_ref() {
+            if location.hostname.contains("rerun.io") {
```
Comment (emilk) [inline crates/re_viewer/src/viewer_analytics.rs:45]: I think we should try to be water tight here

```suggestion
            if location.hostname == "rerun.io" || location.hostname.ends_with(".rerun.io") {
```

**Review by emilk — APPROVED** [review]

File: crates/re_viewer/src/viewer_analytics.rs (hunk 4 above)
Comment (jprochazk) [reply crates/re_viewer/src/viewer_analytics.rs:32]: Added comments along the lines of "We only want to have this on `*.rerun.io` domains" here and also in the hostname check.

**Pull request merged**